.github/copilot-instructions.md

```diff
@@ -0,0 +1,154 @@
+
+### Runtime Validation
+The servers can be started for basic validation:
+```bash
+cd build
+./MasterServer
+```
+- **Server will start but complain about missing client files (this is expected)**
+- **Database connections work with proper configuration in sharedconfig.ini**
+- **For full server testing, LEGO Universe client files are required (not available in this repository)**
+
+### Code Validation
+**ALWAYS validate your changes by**:
+1. Building successfully with no new compilation errors
+2. Running the test suite and confirming all tests pass
+3. Starting MasterServer to verify basic functionality
+4. **Use .editorconfig** - code style uses tabs (width=4), Unix line endings, trailing whitespace removal
+
+## Common Tasks
+
+### Project Structure
+```
+/home/runner/work/DarkflameServer/DarkflameServer/
+├── dAuthServer/          # Authentication server code
+├── dChatServer/          # Chat server code  
+├── dMasterServer/        # Master server (main coordinator)
+├── dWorldServer/         # World/game server code
+├── dCommon/              # Shared common utilities
+├── dDatabase/            # Database abstraction layer
+├── dGame/                # Core game logic, components, behaviors
+├── dScripts/             # Game scripts (NPCs, quests, etc.)
+├── dNet/                 # Network utilities
+├── dPhysics/             # Physics integration
+├── tests/                # Unit tests (GoogleTest)
+├── migrations/           # Database schema migrations
+├── thirdparty/           # External dependencies
+├── build.sh              # Main build script
+├── CMakeVariables.txt    # Build configuration variables
+└── CMakePresets.json     # CMake preset configurations
+```
+
+### Key Files to Know
+- **CMakeVariables.txt**: Build configuration (testing enabled, MariaDB jobs, etc.)
+- **build/sharedconfig.ini**: Database connection, client location, server settings
+- **build/masterconfig.ini**: Master server port and startup configuration
+- **CONTRIBUTING.md**: Code style guidelines and commit message format
+- **docs/Commands.md**: Complete list of in-game server commands
+
+### Build Configuration
+Located in `CMakeVariables.txt`:
+- `ENABLE_TESTING=1` - Unit tests enabled (keep enabled)
+- `MARIADB_CONNECTOR_COMPILE_JOBS=1` - Parallel compilation jobs for MariaDB connector
+- `CDCLIENT_CACHE_ALL=0` - Database caching strategy
+
+### Common Commands Reference
+```bash
+# Build from clean state
+rm -rf build && ./build.sh -j2
+
+# Run specific test
+cd build && ctest -R "TestName" --output-on-failure
+
+# Check which servers were built
+cd build && ls -la *Server
+
+# View build configuration
+cat CMakeVariables.txt
+
+# Check git submodules status
+git submodule status
+```
+
+### Important Notes
+- **Client files are NOT included** - this is only the server emulator
+- **Database can use SQLite or MariaDB** - MariaDB recommended for development
```

Comment:
sqlite is recommended for development since it;s lighter and doesn't require and external service